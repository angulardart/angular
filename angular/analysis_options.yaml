# This is a copy of ../analysis_options.yaml, with implicit-casts: false, and an "exclude: ...".

analyzer:
  strong-mode:
    implicit-casts: false
  exclude:
    - "tools/analyzer_plugin/bin/plugin.dart"
  errors:
    unused_element: error
    unused_import: error
    unused_local_variable: error
    dead_code: error

    # Allow importing .template.dart files without an analyzer error.
    uri_has_not_been_generated: ignore

    # The noise of deprecations isn't worth the effort.
    deprecated_member_use_from_same_package: ignore
linter:
  rules:
    - avoid_empty_else
    - avoid_init_to_null
    - avoid_null_checks_in_equality_operators
    - await_only_futures
    - camel_case_types
    - cancel_subscriptions
    - control_flow_in_finally
    - directives_ordering
    - empty_catches
    - empty_constructor_bodies
    - empty_statements
    - hash_and_equals
    - iterable_contains_unrelated_type
    - library_names
    - library_prefixes
    - list_remove_unrelated_type
    - only_throw_errors
    - overridden_fields
    - package_api_docs
    - package_names
    - package_prefixed_library_names
    - prefer_final_fields
    - prefer_is_not_empty
    - slash_for_doc_comments
    - test_types_in_equals
    - throw_in_finally
    - type_init_formals
    - unnecessary_brace_in_string_interps
    - unrelated_type_equality_checks
    - valid_regexps
