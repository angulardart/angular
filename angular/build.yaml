targets:
  $default:
    builders:
      angular:
        generate_for:
          exclude:
            - "lib/src/build.dart"
            - "lib/src/compiler/**"
            - "lib/src/source_gen/**"

builders:
  angular:
    import: "package:angular/src/build.dart"
    builder_factories:
      - templatePlaceholder
      - templateCompiler
      - stylesheetCompiler
    auto_apply: dependents
    applies_builders:
      - "angular|placeholder_cleanup"
      - "angular|component_source_cleanup"
    # See https://github.com/dart-lang/angular/issues/988.
    is_optional: true
    required_inputs:
      - ".css"
    build_extensions:
      .css:
        - ".css.dart"
        - ".css.shim.dart"
      .dart:
        - ".template.dart"
post_process_builders:
  placeholder_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "placeholderCleanup"
  component_source_cleanup:
    import: "package:angular/src/build.dart"
    builder_factory: "componentSourceCleanup"
    defaults:
      generate_for: ["lib/**"]
      release_options:
        enabled: true
