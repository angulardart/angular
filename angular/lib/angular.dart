/// The primary library for the [AngularDart web framework][AngularDart].
///
/// Import this library as follows:
///
/// ```
/// import 'package:angular/angular.dart';
/// ```
///
/// For help using this library, see the AngularDart documentation:
///
/// * [AngularDart guide][]
/// * [AngularDart cheat sheet][cheatsheet]
///
/// [AngularDart]: https://webdev.dartlang.org/angular
/// [AngularDart guide]: https://webdev.dartlang.org/angular/guide
/// [cheatsheet]: https://webdev.dartlang.org/angular/cheatsheet

library angular;

export 'core.dart';
export 'src/common/common_directives.dart';
export 'src/common/directives.dart';
export 'src/common/pipes.dart';
export 'src/core/angular_entrypoint.dart' show AngularEntrypoint;
export 'src/core/application_tokens.dart' hide APP_ID_RANDOM_PROVIDER;
export 'src/platform/bootstrap.dart';
export 'src/platform/dom/dom_tokens.dart';
export 'src/platform/dom/events/event_manager.dart' show EventManagerPlugin;
