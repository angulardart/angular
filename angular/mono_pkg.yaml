dart:
- stable
- dev

stages:
- analyze:
  - dartanalyzer: --fatal-warnings .
