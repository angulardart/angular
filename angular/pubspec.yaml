name: angular
version: 5.3.0
author: Dart Team <web@dartlang.org>
description: Fast and productive web framework
homepage: https://webdev.dartlang.org/angular
documentation: https://webdev.dartlang.org/angular/api
environment:
  sdk: ">=2.2.1-dev.3.0 <3.0.0"

dependencies:
  # Runtime (or runtime *and* the compiler).
  collection: ^1.14.5
  intl: '>=0.13.0 <0.16.0'
  js: ^0.6.0
  logging: '>=0.9.0 <0.12.0'
  meta: ^1.1.4
  stack_trace: ^1.9.1

  # Compiler. Eventually we want to move these to angular_compiler.
  analyzer: ^0.36.2

  # It is extremely important these two dependencies stay pinned to current.
  angular_ast: 0.5.9
  angular_compiler: 0.4.3
  ##########################################################################

  build: '>=0.12.7 <2.0.0'
  build_config: '>=0.2.6 <0.5.0' # Runtime dependency with no import
  code_builder: '^3.0.1'
  csslib: '>=0.14.5 <0.16.0'
  path: ^1.6.1
  source_gen: ">=0.9.1 <0.10.0"
  source_span: ^1.4.0

# === vvv REMOVE WHEN PUBLISHING vvv ===
dependency_overrides:
  angular_ast:
      path: ../angular_ast
  angular_compiler:
      path: ../angular_compiler
# === ^^^ REMOVE WHEN PUBLISHING ^^^ ===
