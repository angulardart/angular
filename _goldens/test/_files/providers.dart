import 'package:angular/angular.dart';

@Component(
  selector: 'providers',
  template: 'Hello',
  providers: [
    MyTypeAnnotation,
    MyInjectableTypeAnnotation,
    Provider(
      MyUseFactory,
      useFactory: ProvidersComponent.createService,
    ),
    Provider(
      MyUseClass,
      useClass: MyUseClass,
    ),
    Provider(
      MyUseExisting,
      useExisting: MyUseClass,
    ),
    Provider(
      MyUseExistingNested,
      useExisting: MyUseExisting,
    ),
    Provider(
      MyUseValue,
      useValue: MyUseValue('Andrew'),
    ),
    Provider(
      useValueString,
      useValue: 'foo',
    ),
    Provider(useValueList, useValue: [
      MyUseValue('Andrew'),
      MyUseValue('Matan'),
      MyUseValue.named(optional: true)
    ]),
    Provider(useValueMap, useValue: {
      'Andrew': MyUseValue('Andrew'),
      'Matan': MyUseValue('Matan')
    }),
    Provider(OpaqueToken('useEnums'), useValue: MyEnum.first),
    Provider(XsrfToken(), useValue: 'ABC123'),
  ],
  viewProviders: [Provider(MyUseValue, useValue: MyUseValue('Matan'))],
)
class ProvidersComponent {
  static MyUseFactory createService(NgZone ngZone, {bool optional}) =>
      MyUseFactory();
}

class MyTypeAnnotation {}

@Injectable()
class MyInjectableTypeAnnotation {}

class MyUseExisting {}

class MyUseClass implements MyUseExisting {}

class MyUseExistingNested implements MyUseExisting {}

class MyUseFactory {}

class MyUseValue {
  const MyUseValue(String name);

  const MyUseValue.named({bool optional});
}

class MyMulti {}

const useValueString = OpaqueToken('useValueString');
const useValueList = OpaqueToken('useValueList');
const useValueMap = OpaqueToken('useValueMap');

enum MyEnum { first, second }

class XsrfToken extends OpaqueToken<String> {
  const XsrfToken();
}
