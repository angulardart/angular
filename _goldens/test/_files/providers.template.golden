// **************************************************************************
// Generator: AngularDart Compiler
// **************************************************************************

import 'providers.dart';
export 'providers.dart';
import 'package:angular/src/di/reflector.dart' as _ngRef;
import 'package:angular/angular.template.dart' as _ref0;
import 'package:angular/src/core/linker/app_view.dart';
import 'providers.dart' as import1;
import 'package:angular/src/core/linker/style_encapsulation.dart' as import2;
import 'package:angular/src/core/linker/view_type.dart' as import3;
import 'package:angular/src/core/change_detection/change_detection.dart';
import 'dart:html' as import5;
import 'package:angular/src/runtime.dart' as import6;
import 'package:angular/angular.dart';
import 'package:angular/src/runtime/dom_helpers.dart' as import8;
import 'package:angular/src/di/errors.dart' as import9;
import 'package:angular/src/core/zone/ng_zone.dart' as import10;
import 'package:angular/src/core/di/opaque_token.dart' as import11;
import 'providers.dart' as import12;

final List<dynamic> styles$ProvidersComponent = const [];

class ViewProvidersComponent0 extends AppView<import1.ProvidersComponent> {
  static import2.ComponentStyles _componentStyles;
  ViewProvidersComponent0(AppView<dynamic> parentView, int parentIndex) : super(import3.ViewType.component, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    rootEl = import5.document.createElement('providers');
    initComponentStyles();
  }
  static String get _debugComponentUrl {
    return (import6.isDevMode ? 'asset:_goldens/test/_files/providers.dart' : null);
  }

  @override
  ComponentRef<import1.ProvidersComponent> build() {
    final _rootEl = rootEl;
    final import5.HtmlElement parentRenderNode = initViewRoot(_rootEl);
    final _text_0 = import8.appendText(parentRenderNode, 'Hello');
    init(const [], null);
  }

  @override
  void initComponentStyles() {
    var styles = _componentStyles;
    if (identical(styles, null)) {
      (_componentStyles = (styles = (_componentStyles = import2.ComponentStyles.unscoped(styles$ProvidersComponent, _debugComponentUrl))));
    }
    componentStyles = styles;
  }
}

AppView<import1.ProvidersComponent> viewFactory_ProvidersComponent0(AppView<dynamic> parentView, int parentIndex) {
  return ViewProvidersComponent0(parentView, parentIndex);
}

const ComponentFactory<import1.ProvidersComponent> _ProvidersComponentNgFactory = const ComponentFactory('providers', viewFactory_ProvidersComponentHost0);
ComponentFactory<import1.ProvidersComponent> get ProvidersComponentNgFactory {
  return _ProvidersComponentNgFactory;
}

final List<dynamic> styles$ProvidersComponentHost = const [];

class _ViewProvidersComponentHost0 extends AppView<import1.ProvidersComponent> {
  ViewProvidersComponent0 _compView_0;
  import1.ProvidersComponent _ProvidersComponent_0_5;
  import1.MyTypeAnnotation __MyTypeAnnotation_0_6;
  import1.MyInjectableTypeAnnotation __MyInjectableTypeAnnotation_0_7;
  dynamic __MyUseFactory_0_8;
  import1.MyUseClass __MyUseClass_0_9;
  import1.MyUseValue __MyUseValue_0_12;
  String __useValueString_0_13;
  List<dynamic> __useValueList_0_14;
  Map<String, dynamic> __useValueMap_0_15;
  dynamic __useEnums_0_16;
  String __XsrfToken_0_17;
  _ViewProvidersComponentHost0(AppView<dynamic> parentView, int parentIndex) : super(import3.ViewType.host, {}, parentView, parentIndex, ChangeDetectionStrategy.CheckAlways) {
    initComponentStyles();
  }
  import1.MyTypeAnnotation get _MyTypeAnnotation_0_6 {
    if ((__MyTypeAnnotation_0_6 == null)) {
      (__MyTypeAnnotation_0_6 = import1.MyTypeAnnotation());
    }
    return __MyTypeAnnotation_0_6;
  }

  import1.MyInjectableTypeAnnotation get _MyInjectableTypeAnnotation_0_7 {
    if ((__MyInjectableTypeAnnotation_0_7 == null)) {
      (__MyInjectableTypeAnnotation_0_7 = import1.MyInjectableTypeAnnotation());
    }
    return __MyInjectableTypeAnnotation_0_7;
  }

  dynamic get _MyUseFactory_0_8 {
    if ((__MyUseFactory_0_8 == null)) {
      (__MyUseFactory_0_8 = (import6.isDevMode
          ? import9.debugInjectorWrap(import1.MyUseFactory, () {
              return import1.ProvidersComponent.createService(this.injectorGet(import10.NgZone, viewData.parentIndex));
            })
          : import1.ProvidersComponent.createService(this.injectorGet(import10.NgZone, viewData.parentIndex))));
    }
    return __MyUseFactory_0_8;
  }

  import1.MyUseClass get _MyUseClass_0_9 {
    if ((__MyUseClass_0_9 == null)) {
      (__MyUseClass_0_9 = import1.MyUseClass());
    }
    return __MyUseClass_0_9;
  }

  import1.MyUseValue get _MyUseValue_0_12 {
    if ((__MyUseValue_0_12 == null)) {
      (__MyUseValue_0_12 = const import1.MyUseValue('Matan'));
    }
    return __MyUseValue_0_12;
  }

  String get _useValueString_0_13 {
    if ((__useValueString_0_13 == null)) {
      (__useValueString_0_13 = 'foo');
    }
    return __useValueString_0_13;
  }

  List<dynamic> get _useValueList_0_14 {
    if ((__useValueList_0_14 == null)) {
      (__useValueList_0_14 = const [import1.MyUseValue('Andrew'), import1.MyUseValue('Matan'), import1.MyUseValue.named(optional: true)]);
    }
    return __useValueList_0_14;
  }

  Map<String, dynamic> get _useValueMap_0_15 {
    if ((__useValueMap_0_15 == null)) {
      (__useValueMap_0_15 = const {'Andrew': import1.MyUseValue('Andrew'), 'Matan': import1.MyUseValue('Matan')});
    }
    return __useValueMap_0_15;
  }

  dynamic get _useEnums_0_16 {
    if ((__useEnums_0_16 == null)) {
      (__useEnums_0_16 = import1.MyEnum.first);
    }
    return __useEnums_0_16;
  }

  String get _XsrfToken_0_17 {
    if ((__XsrfToken_0_17 == null)) {
      (__XsrfToken_0_17 = 'ABC123');
    }
    return __XsrfToken_0_17;
  }

  @override
  ComponentRef<import1.ProvidersComponent> build() {
    _compView_0 = ViewProvidersComponent0(this, 0);
    rootEl = _compView_0.rootEl;
    _ProvidersComponent_0_5 = import1.ProvidersComponent();
    _compView_0.create(_ProvidersComponent_0_5, projectableNodes);
    init0(rootEl);
    return ComponentRef(0, this, rootEl, _ProvidersComponent_0_5);
  }

  @override
  dynamic injectorGetInternal(dynamic token, int nodeIndex, dynamic notFoundResult) {
    if ((identical(token, import1.MyTypeAnnotation) && (0 == nodeIndex))) {
      return _MyTypeAnnotation_0_6;
    }
    if ((identical(token, import1.MyInjectableTypeAnnotation) && (0 == nodeIndex))) {
      return _MyInjectableTypeAnnotation_0_7;
    }
    if ((identical(token, import1.MyUseFactory) && (0 == nodeIndex))) {
      return _MyUseFactory_0_8;
    }
    if ((((identical(token, import1.MyUseClass) || identical(token, import1.MyUseExisting)) || identical(token, import1.MyUseExistingNested)) && (0 == nodeIndex))) {
      return _MyUseClass_0_9;
    }
    if ((identical(token, import1.MyUseValue) && (0 == nodeIndex))) {
      return _MyUseValue_0_12;
    }
    if ((identical(token, const import11.OpaqueToken<dynamic>('useValueString')) && (0 == nodeIndex))) {
      return _useValueString_0_13;
    }
    if ((identical(token, const import11.OpaqueToken<dynamic>('useValueList')) && (0 == nodeIndex))) {
      return _useValueList_0_14;
    }
    if ((identical(token, const import11.OpaqueToken<dynamic>('useValueMap')) && (0 == nodeIndex))) {
      return _useValueMap_0_15;
    }
    if ((identical(token, const import11.OpaqueToken<dynamic>('useEnums')) && (0 == nodeIndex))) {
      return _useEnums_0_16;
    }
    if ((identical(token, const import12.XsrfToken()) && (0 == nodeIndex))) {
      return _XsrfToken_0_17;
    }
    return notFoundResult;
  }

  @override
  void detectChangesInternal() {
    _compView_0.detectChanges();
  }

  @override
  void destroyInternal() {
    _compView_0.destroy();
  }
}

AppView<import1.ProvidersComponent> viewFactory_ProvidersComponentHost0(AppView<dynamic> parentView, int parentIndex) {
  return _ViewProvidersComponentHost0(parentView, parentIndex);
}

var _visited = false;
void initReflector() {
  if (_visited) {
    return;
  }
  _visited = true;

  _ngRef.registerComponent(ProvidersComponent, ProvidersComponentNgFactory);
  _ngRef.registerFactory(MyInjectableTypeAnnotation, () => new MyInjectableTypeAnnotation());
  _ref0.initReflector();
}
